libraries/libfc/src/log/logger_config.cpp

```diff
@@ -133,26 +136,22 @@ namespace fc {
    }
 
    static thread_local std::string thread_name;
-   void set_os_thread_name( const std::string& name ) {
-#ifdef FC_USE_PTHREAD_NAME_NP
-      pthread_setname_np( pthread_self(), name.c_str() );
-#endif
-   }
+
    void set_thread_name( const std::string& name ) {
       thread_name = name;
+#if defined(__linux__) || defined(__FreeBSD__)
+      pthread_setname_np( pthread_self(), name.c_str() );
+#elif defined(__APPLE__)
+      pthread_setname_np( name.c_str() );
+#endif
    }
    const std::string& get_thread_name() {
-      if( thread_name.empty() ) {
-#ifdef FC_USE_PTHREAD_NAME_NP
-         char thr_name[64];
-         int rc = pthread_getname_np( pthread_self(), thr_name, 64 );
-         if( rc == 0 ) {
-            thread_name = thr_name;
+      if(thread_name.empty()) {
+         try {
+            thread_name = boost::dll::program_location().filename().generic_string();
+         } catch (...) {
+            thread_name = "unknown";
          }
-#else
-         static int thread_count = 0;
-         thread_name = std::string( "thread-" ) + std::to_string( thread_count++ );
```

Comment:
On mac we used to always get the `thread-n` version here. Does this now provide the actual thread name on mac?

Reply:
yes; really for any platform the thread names should be logged identically now since `get_thread_name()` just pulls from the `static thread_local std::string thread_name` -- that is, no platform specific code is invoked for thread names for logging purposes.

But on macos & freebsd we've also gained the thread names being set at the platform level, and I've confirmed on macos they do show in the debugger.